mobile-center-crashes/Crashes.js

```diff
@@ -76,29 +81,31 @@ let Crashes = {
         }
         getErrorAttachmentsMethod = listenerMap.getErrorAttachments;
         RNCrashes.getUnprocessedCrashReports()
-        .then((reports) => {
-            const filteredReportIds = [];
-            reports.forEach((report) => {
-                if (!listenerMap.shouldProcess ||
-                    listenerMap.shouldProcess(report)) {
-                    filteredReports.push(report);
-                    filteredReportIds.push(report.id);
+            .then((reports) => {
+                if (reports.length > 0) {
+                    const filteredReportIds = [];
+                    reports.forEach((report) => {
+                        if (!listenerMap.shouldProcess ||
+                            listenerMap.shouldProcess(report)) {
+                            filteredReports.push(report);
+                            filteredReportIds.push(report.id);
+                        }
+                    });
+                    RNCrashes.sendCrashReportsOrAwaitUserConfirmationForFilteredIds(filteredReportIds).then((alwaysSend) => {
+                        if (alwaysSend) {
+                            Helper.sendErrorAttachments(filteredReports);
+                        } else if (!listenerMap.shouldAwaitUserConfirmation || !listenerMap.shouldAwaitUserConfirmation()) {
```

Comment:
not sure why you need `!listenerMap.shouldAwaitUserConfirmation` here?

Reply:
Because the callback can be undefined since it's optional to define it.

Reply:
Can you document this subtlety? It's not very obvious.

Reply:
We do that a lot for every callback, it's very common for JS callbacks.